components/filter2/dropdown_group/view.coffee

```diff
@@ -35,7 +35,7 @@ module.exports = class DropdownGroupView extends Backbone.View
 
   toggleForSale: ->
     if @params.get('for_sale') is 'true'
-      @params.set 'for_sale', 'false'
+      @params.set 'for_sale', null
```

Comment:
One effect of this is that it sets the query string in the URLs (on browser and to API) to `...&for_sale=&...`. I haven't seen any side effects of this, but it clutters the URLs with a blank parameter. Trimming out empty parameters from the query strings will be a deeper change though — and probably more appropriate for its own PR.